src/test/regress/expected/subquery_and_cte.out

```diff
@@ -471,6 +478,524 @@ END;
+		ORDER BY
+			user_id ASC
+		LIMIT
+			10
+		)
+		UNION ALL
+		(SELECT
+			value_1
+		FROM
+			users_table
+		ORDER BY
+			value_1 ASC
+		LIMIT
+			10
+		)
+	) as SUB_TABLE
+WHERE
+	user_id
+IN
+	(
+	WITH event_id AS (
+		SELECT
+			user_id as events_user_id, time as events_time, event_type
+		FROM
+			events_table
+	)
+	SELECT
+		events_user_id
+	FROM
+		event_id
+	ORDER BY
+		events_user_id
+	LIMIT
+		10
+	);
+ user_id 
+---------
+       1
+       1
+       1
+       1
+       1
+       1
+       1
+(7 rows)
+
+-- Complex target list in WHERE clause
+SELECT
+	COUNT(*)
+FROM
+	(SELECT
+		user_id as events_user_id, time as events_time, event_type
+	FROM
+		events_table
+	LIMIT
+		10
+	) as SUB_TABLE
+WHERE
+	events_user_id
+<= (
+	SELECT
+		max(abs(value_1 * 1) + mod(value_1, 3)) as val_1
+	FROM
+		users_table
+);
+ count 
+-------
+    10
+(1 row)
+
+-- DISTINCT clause in WHERE
+SELECT
+	COUNT(*)
+FROM
+	(SELECT
+		user_id as events_user_id, time as events_time, event_type
+	FROM
+		events_table
+	LIMIT
+		10
+	) as SUB_TABLE
+WHERE
+	events_user_id
+IN (
+	SELECT
+		distinct value_1
+	FROM
+		users_table
+);
+ count 
+-------
+    10
+(1 row)
+
+-- AND in WHERE clause
+SELECT
+	COUNT(*)
+FROM
+	(SELECT
+		user_id as events_user_id, time as events_time, event_type
+	FROM
+		events_table
+	LIMIT
+		10
+	) as SUB_TABLE
+WHERE
+	events_user_id
+> (
+	SELECT
+		min(value_1)
+	FROM
+		users_table
+)
+AND
+	events_user_id
+< (
+	SELECT
+		max(value_2)
+	FROM
+		users_table
+);
+ count 
+-------
+     4
+(1 row)
+
+-- Planning subqueries in WHERE clause in CTE recursively
+WITH cte AS (
+	SELECT
+		*
+	FROM
+		(SELECT
+			*
+		FROM
+			users_table
+		ORDER BY
+			user_id ASC,
+			value_2 DESC
+		LIMIT
+			10
+		) as sub_table
+	WHERE
+		user_id
+	IN
+		(SELECT
+			value_2
+		FROM
+			events_table
+		)
+)
+SELECT
+	COUNT(*)
+FROM
+	cte;
+ count 
+-------
+    10
+(1 row)
+
+-- Planing subquery in WHERE clause in FROM clause of a subquery recursively
+SELECT
+	COUNT(*)
+FROM
+	(SELECT
+		*
+	FROM
+		(SELECT
+			*
+		FROM
+			users_table
+		ORDER BY
+			user_id ASC,
+			value_2 DESC
+		LIMIT
+			10
+		) as sub_table_1
+	WHERE
+		user_id
+	IN
+		(SELECT
+			value_2
+		FROM
+			events_table
+		)
+	) as sub_table_2;
+ count 
+-------
+    10
+(1 row)
+
+-- Recurring table in the FROM clause of a subquery in the FROM clause
+-- Recurring table is created by joining a two recurrign table
+SELECT
+	SUM(user_id)
+FROM
+	(SELECT
+		*
+	FROM
+		(SELECT
+			user_id
+		FROM
+			users_table
+		ORDER BY
+			user_id
+		LIMIT 10) as t1
+		INNER JOIN
+		(SELECT
+			user_id as user_id_2
+		FROM
+			users_table
+		ORDER BY
+			user_id
+		LIMIT
+			10) as t2
+		ON
+			t1.user_id = t2.user_id_2
+		WHERE
+			t1.user_id
+		IN
+			(SELECT
+				value_2
+			FROM
+				events_table)
+	) as t3
+WHERE
+	user_id
+>
+	(SELECT
+		min(value_3)
+	FROM
+		events_table);
+ sum 
+-----
+  67
+(1 row)
+
+-- Same example with the above query, but now check the rows with EXISTS
+SELECT
```

Comment:
We should move all the tests into a new file, maybe: `subquery_in_where.sql` or such